src/WPF/ArcGISRuntime.WPF.Samples/Samples/Map/SearchPortalMaps/SearchPortalMaps.xaml.cs

```diff
@@ -0,0 +1,418 @@
+
+            // Type of token authentication to use
+            portalServerInfo.TokenAuthenticationType = TokenAuthenticationType.OAuthImplicit;
+
+            // Define the OAuth information
+            OAuthClientInfo oAuthInfo = new OAuthClientInfo
+            {
+                ClientId = AppClientId,
+                RedirectUri = new Uri(OAuthRedirectUrl)
+            };
+            portalServerInfo.OAuthClientInfo = oAuthInfo;
+
+            // Get a reference to the (singleton) AuthenticationManager for the app
+            AuthenticationManager thisAuthenticationManager = AuthenticationManager.Current;
+
+            // Register the ArcGIS Online server information with the AuthenticationManager
+            thisAuthenticationManager.RegisterServer(portalServerInfo);
+
+            // Use the OAuthAuthorize class in this project to create a new web view to show the login UI
+            thisAuthenticationManager.OAuthAuthorizeHandler = new OAuthAuthorize();
+
+            // Create a new ChallengeHandler that uses a method in this class to challenge for credentials
+            thisAuthenticationManager.ChallengeHandler = new ChallengeHandler(CreateCredentialAsync);
+        }
+
+        // ChallengeHandler function that will be called whenever access to a secured resource is attempted
+        public async Task<Credential> CreateCredentialAsync(CredentialRequestInfo info)
+        {
+            Credential credential = null;
+
+            try
+            {
+                // IOAuthAuthorizeHandler will challenge the user for OAuth credentials
+                credential = await AuthenticationManager.Current.GenerateCredentialAsync(info.ServiceUri);
+            }
+            catch (Exception ex)
+            {
+                // Exception will be reported in calling function
+                throw (ex);
+            }
+
+            return credential;
+        }
+
+        private void SaveOAuthSettingsClicked(object sender, RoutedEventArgs e)
+        {
+            // Settings were provided, update the configuration settings for OAuth authorization
+            AppClientId = ClientIdTextBox.Text.Trim();
+            OAuthRedirectUrl = RedirectUrlTextBox.Text.Trim();
+
+            // Update authentication manager with the OAuth settings
+            UpdateAuthenticationManager();
+
+            // Hide the OAuth input, show the search UI
+            OAuthSettingsGrid.Visibility = Visibility.Collapsed;
+            SearchUI.Visibility = Visibility.Visible;
+        }
+
+        private void CancelOAuthSettingsClicked(object sender, RoutedEventArgs e)
+        {
+            // Warn that browsing user's ArcGIS Online maps won't be available without OAuth settings
+            var noAuth = MessageBox.Show("Without OAuth settings, you will not be able to browse maps from your ArcGIS Online account.", "No OAuth Settings", MessageBoxButton.OKCancel) == MessageBoxResult.OK;
```

Comment:
Can this line be broken across two lines? The line is so long that the user will have to do scrolling in the code example window of the LibRefAPI and GuideDoc samples.